src/main/java/net/dv8tion/jda/api/entities/GuildScheduledEvent.java

```diff
@@ -0,0 +1,393 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.dv8tion.jda.api.entities;
+
+import net.dv8tion.jda.annotations.Incubating;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.managers.GuildScheduledEventManager;
+import net.dv8tion.jda.api.requests.RestAction;
+
+import javax.annotation.CheckReturnValue;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.time.OffsetDateTime;
+
+/**
+ * A class representing a guild scheduled event (the ones that show up under the events tab in the Official Discord Client).
+ * These events should not be confused with {@link net.dv8tion.jda.api.events Gateway Events},
+ * which are fired by Discord whenever something interesting happens
+ * (ie., a {@link net.dv8tion.jda.api.events.message.MessageDeleteEvent MessageDeleteEvent} gets fired whenever a message gets deleted).
+ */
+public interface GuildScheduledEvent extends ISnowflake, Comparable<GuildScheduledEvent>
+{
+    /**
+     * The maximum allowed length for an event's name.
+     */
+    int MAX_NAME_LENGTH = 100;
+
+    /**
+     * The maximum allowed length for an event's description.
+     */
+    int MAX_DESCRIPTION_LENGTH = 1000;
+
+    /**
+     * The name of the event.
+     *
+     * @return The name
+     */
+    @Nonnull
+    String getName();
+
+    /**
+     * The description of the event.
+     *
+     * @return The description, or {@code null} if none is specified
+     */
+    @Nullable
+    String getDescription();
+
+    /**
+     * The user who originally created the event.
+     * <p> A {@code null} may be returned if user has deleted their account, the {@link User} object has not yet been cached
+     * (lazy-loading) or if the event was created before Discord started keeping track of event creators on October 21st, 2021. {@link #hasCreator()} may be used to check if the
+     * event has a creator associated with it.
+     * <p> The creator may additionally be retrieved using {@link #retrieveCreator()} if the event {@link #hasCreator() has a creator}
+     * but a {@code null} value is returned due to the {@link User} not being cached.
+     *
+     * @return Possiblly-null {@link User} object representing the event's creator.
+     *
+     * @see    #hasCreator()
+     * @see    #getCreatorId()
+     * @see    #getCreatorIdLong()
+     * @see    #retrieveCreator()
+     */
+    @Nullable
+    User getCreator();
+
+    // Todo: Document
+    @Nonnull
+    @CheckReturnValue
+    RestAction<User> retrieveCreator();
+
+    /**
+     * The ID of the user who originally created this event.
+     * <p> This method may return -1 if the event was created before Discord started keeping track of event creators on
+     * October 21st, 2021.
+     *
+     * @return The ID of the user who created this event, or -1 if no user is associated with creating this event.
```

Comment:
Afaik JDA provides `0` & not `-1` for missing IDs. Does discord provide `-1`?

Reply:
JDA has never used `0` for unknown, always `-1`.